Source/ASCollectionView.mm

```diff
@@ -1467,6 +1486,26 @@ - (void)_beginBatchFetching
   }
 }
 
+#pragma mark - ASCollectionViewLayoutCalculatorProviding
+
+- (id<ASCollectionViewLayoutCalculating>)collectionViewLayoutCalculator
+{
+  return _collectionViewLayoutFlags.providesLayoutCalculator ? [(id<ASCollectionViewLayoutCalculatorProviding>)self.collectionViewLayout collectionViewLayoutCalculator] : nil;
+}
+
+- (CGSize)sizeForCalculatingCollectionViewLayout
+{
+  return _collectionViewLayoutFlags.providesLayoutCalculator ? [(id<ASCollectionViewLayoutCalculatorProviding>)self.collectionViewLayout sizeForCalculatingCollectionViewLayout] : CGSizeZero;
+}
+
+#pragma mark - ASCollectionViewLayoutCalculatorResultConsuming
```

Comment:
I like the idea of dropping "View" in these names as well. I'm about to experiment letting `ASTableView` to provide a calculator but doesn't consume the result. The idea is to entirely remove the layout responsibility from ASDataController and let calculators handle it.